ledger/src/blockstore_processor.rs

```diff
@@ -1081,7 +1081,9 @@ fn process_bank_0(
         None,
         &mut ExecuteTimings::default(),
     )
-    .expect("processing for bank 0 must succeed");
+    .expect(
+        "Failed to process bank 0 from ledger. Did you forget to specify a snapshot directory?",
```

Comment:
mis-specifying the snapshot dir isn't the only path to get here.  Similarly, a validator can be started with `--no-snapshot-fetch` and no local snapshots.  How far away is the snapshot dir from here?  A more generic message, similar to the other one would make sense here.

Reply:
I tweaked the error message a little.

Including the directory searched for the snapshot  in this panic feels clunky. We can get here by not finding a snapshot (which produces the warning that includes the snapshot directory), or there being no snapshot config. But both of those conditions are recoverable if the local ledger starts at genesis.

The snapshot directory is 2 levels up in the call stack. I can include that info in the panic by returning an error up those two levels but passing an unrecoverable error around to trigger a panic somewhere else feels cumbersome.

The current behavior will produce logs that look like this for both the ledger tool run without a snapshot and the validator missing a local snapshot and run with `--no-snapshot-fetch`.

```[2022-05-20T20:53:06.765280672Z WARN  solana_ledger::bank_forks_utils] No snapshot package found in directory: "/mnt/nvme1n1/ledger"; will load from genesis
[2022-05-20T20:53:06.765288777Z INFO  solana_ledger::bank_forks_utils] Processing ledger from genesis
[2022-05-20T20:53:06.838563382Z INFO  solana_metrics::metrics] datapoint: bank-timestamp get_timestamp_estimate_us=1i
[2022-05-20T20:53:06.838577830Z INFO  solana_metrics::metrics] datapoint: bank-timestamp-correction slot=0i from_genesis=1580834132i corrected=0i ancestor_timestamp=0i
[2022-05-20T20:53:06.838723908Z INFO  solana_ledger::blockstore_processor] processing ledger for slot 0...
thread 'main' panicked at 'Failed to process bank 0 from ledger. Did you forget to provide a snapshot?: InvalidBlock(Incomplete)', ledger/src/blockstore_processor.rs:1120:6
stack backtrace:```